File: samples/vbnet/blob_test/blob_test/Form1.vb
Option Strict On
Imports MySql.Data.MySqlClient
Imports System.IO
Public Class Form1
  Dim myconn As MySqlConnection
  Dim com As MySqlCommand
  Dim id As Integer

  ' connect to MySQL server, create testpic table
  Private Sub btnConnect_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnConnect.Click
    Try
      myconn = New MySqlConnection( _
          "Data Source=192.168.80.128;Initial Catalog=mylibrary;" + _
          "User ID=root;PWD=uranus")
      myconn.Open()
      com = New MySqlCommand( _
        "CREATE TABLE IF NOT EXISTS testpic " + _
        "  (id INT NOT NULL AUTO_INCREMENT, pic MEDIUMBLOB, PRIMARY KEY (id))", _
        myconn)
      com.ExecuteNonQuery()
      'enable Save button
      btnSave.Enabled = True
    Catch myerror As MySqlException
      MsgBox("Database connection error: " & myerror.Message)
      Me.Close()
    End Try
  End Sub

  ' load file, save it as blob in database
  Private Sub btnSave_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSave.Click
    Dim fs As FileStream
    Dim bindata As Byte()
    Dim picpara As MySqlParameter

    ' prepare INSERT command
    com = New MySqlCommand( _
      "INSERT INTO testpic (pic) VALUES(?pic)", myconn)
    picpara = com.Parameters.Add("?pic", MySqlDbType.MediumBlob)
    com.Prepare()

    ' read JPEG file into Byte array
    fs = New FileStream("test.jpg", FileMode.Open, FileAccess.Read)
    ReDim bindata(CInt(fs.Length))
    fs.Read(bindata, 0, CInt(fs.Length))
    fs.Close()

    ' call INSERT command, use Byte array as parameter
    picpara.Value = bindata
    com.ExecuteNonQuery()

    ' retrieve id of new record
    com.CommandText = "SELECT LAST_INSERT_ID()"
    id = CInt(com.ExecuteScalar())

    ' enable load button
    btnLoad.Enabled = True
  End Sub

  ' load BLOB from database, save as file, show as image
  Private Sub btnLoad_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnLoad.Click
    Dim fs As FileStream
    Dim bindata As Byte()
    Dim ms As New MemoryStream

    com = New MySqlCommand("SELECT pic FROM testpic WHERE id=" & id, myconn)
    bindata = CType(com.ExecuteScalar(), Byte())

    ' show JPEG file
    ms.Write(bindata, 0, bindata.Length)
    PictureBox1.Image = New Bitmap(ms)

    ' save new JPEG file
    fs = New FileStream("test1.jpg", FileMode.Create, FileAccess.Write)
    ms.WriteTo(fs)
    fs.Close()
  End Sub

  ' delete testpic table, end
  Private Sub btnEnd_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnEnd.Click
    com = New MySqlCommand( _
      "DROP TABLE IF EXISTS testpic ", myconn)
    com.ExecuteNonQuery()
    Me.Close()
  End Sub
End Class

File: samples/csharp/blob_test/blob_test/Form1.cs
#region Using directives

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Windows.Forms;
using System.IO;
using MySql.Data.MySqlClient;

#endregion

namespace blob_test {
  partial class Form1 : Form {
    public Form1() {
      InitializeComponent();
    }

    MySqlConnection myconn;
    MySqlCommand com;
    int id;

    // connect to MySQL server, create testpic table
    private void btnConnect_Click(object sender, EventArgs e) {
      try {
        myconn = new MySqlConnection(
          "Data Source=192.168.80.128;Initial Catalog=mylibrary;" +
          "User ID=root;PWD=uranus");
        myconn.Open();
        com = new MySqlCommand(
          "CREATE TABLE IF NOT EXISTS testpic " + 
          "  (id INT NOT NULL AUTO_INCREMENT, pic MEDIUMBLOB, PRIMARY KEY (id))", 
          myconn);
        com.ExecuteNonQuery();
        btnSave.Enabled = true;
      }
      catch (Exception ex) {
        Console.WriteLine(ex.Message);
        this.Close();
      }
    }

    // load test1.jpg, save it in BLOB
    private void btnSave_Click(object sender, EventArgs e) {
      byte[] bindata;
      FileStream fs;
      MySqlParameter picpara;

      // prepare INSERT command
      com = new MySqlCommand( 
        "INSERT INTO testpic (pic) VALUES(?pic)", myconn);
      picpara = com.Parameters.Add("?pic", MySqlDbType.MediumBlob);
      com.Prepare();

      // read JPEG file into Byte array
      fs = new FileStream("test.jpg", FileMode.Open, FileAccess.Read);
      bindata = new byte[fs.Length];
      fs.Read(bindata, 0, (int)fs.Length);
      fs.Close();

      // call INSERT command, use Byte array as parameter
      picpara.Value = bindata;
      com.ExecuteNonQuery();

      // retrieve id of new record
      com.CommandText = "SELECT LAST_INSERT_ID()";
      id = (int)(long)com.ExecuteScalar();

      // enable load button
      btnLoad.Enabled = true;
    }

    // load data from BLOB, save in test1.jpg, show in PictureBox1
    private void btnLoad_Click(object sender, EventArgs e) {
      byte[] bindata;
      FileStream fs;
      MemoryStream ms;

      com = new MySqlCommand("SELECT pic FROM testpic WHERE id=" + id.ToString(), myconn);
      bindata = (byte[]) com.ExecuteScalar();

      // show JPEG file
      ms = new MemoryStream();
      ms.Write(bindata, 0, bindata.Length);
      PictureBox1.Image = new Bitmap(ms);

      // save new JPEG file
      fs = new FileStream("test1.jpg", FileMode.Create, FileAccess.Write);
      ms.WriteTo(fs);
      fs.Close();
    }
    
    private void btnEnd_Click(object sender, EventArgs e) {
      com = new MySqlCommand( 
        "DROP TABLE IF EXISTS testpic ", myconn);
      com.ExecuteNonQuery();
      this.Close();
    }

  }
}